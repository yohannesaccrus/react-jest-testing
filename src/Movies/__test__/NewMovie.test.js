import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import NewMovie from '../NewMovie'

afterEach(cleanup)

test('<NewMovie/>', () => {
  const { debug, getByTestId, queryByTestId, container } = render(<NewMovie />)
  const pageTitle = getByTestId('page-title')
  const movieForm = queryByTestId('movie-form')

  expect(pageTitle.textContent).toBe('New Movie')
  expect(movieForm).toBeTruthy()
  expect(container.firstChild).toMatchSnapshot()
})
