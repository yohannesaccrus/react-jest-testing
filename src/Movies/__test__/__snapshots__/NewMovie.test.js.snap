// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie/> 1`] = `
<div>
  <h2
    data-testid="page-title"
  >
    New Movie
  </h2>
  <div>
    <form
      data-testid="movie-form"
    >
      <input
        type="text"
        id="name"
        data-testid="input-name"
      />
      <button
        data-testid="submit-btn"
      >
        submit
      </button>
    </form>
  </div>
</div>
`;
